tests/questionpy_server/worker/impl/test_subprocess.py

```diff
@@ -26,3 +38,45 @@ async def test_should_apply_limits(pool: WorkerPool) -> None:
         soft, hard = psutil_process.rlimit(resource.RLIMIT_AS)
 
     assert soft == hard == 200 * MiB
+
+
+@contextmanager
+def _make_get_manifest_busy_wait() -> Iterator[None]:
+    def busy_wait(self: WorkerManager) -> None:
+        wait_until = process_time() + 10
+        while wait_until > process_time():
+            pass
+
+    with patch.object(WorkerManager, "bootstrap", busy_wait):
+        yield
+
+
+async def test_should_raise_cpu_timout_error(pool: WorkerPool) -> None:
+    with patch_worker_pool(pool, _make_get_manifest_busy_wait):
+        with pytest.raises(WorkerStartError) as exc_info, patch.object(BaseWorker, "_init_worker_timeout", 0.05):
+            async with pool.get_worker(PACKAGE, 1, 1):
+                pass
+        assert isinstance(exc_info.value.__cause__, WorkerCPUTimeLimitExceededError)
+
+
+@contextmanager
+def _make_get_manifest_sleep() -> Iterator[None]:
+    def _sleep(self: WorkerManager) -> None:
+        sleep(10)
+
+    with patch.object(WorkerManager, "bootstrap", _sleep):
+        yield
+
+
+async def test_should_raise_real_timout_error(pool: WorkerPool) -> None:
+    with patch_worker_pool(pool, _make_get_manifest_sleep):
+        # The timeout should not be too short, because the Python interpreter also needs some time to start up, which
+        # is accounted for the init worker step. Otherwise, a WorkerCPUTimeLimitExceededError is raised.
+        with (
+            pytest.raises(WorkerStartError) as exc_info,
+            patch.object(BaseWorker, "_init_worker_timeout", 0.6),
+            patch.object(LimitTimeUsageMixin, "_real_time_limit_factor", 0.1),
```

Comment:
Ist es nötig, hier das Echtzeitlimit niedriger zu stellen als das CPU-Zeit-Limit? Das finde ich etwas verwirrend mit Blick auf den Kommentar `CPU-time is always less or equal to real time (when single-threaded).`.

Reply:
Stimmt, weniger als 1,0 bringt keinen Effekt durch das initiale Warten.